napari/_qt/qt_main_window.py

```diff
@@ -279,6 +294,8 @@ def _save_current_window_settings(self):
 
     def close(self, quit_app=False, confirm_need=False):
         """Override to handle closing app or just the window."""
+        if hasattr(self.status_throttler, "_timer"):
```

Comment:
I guess `self.status_throttler.cancel()` isn't enough here right? Would be nice to have a better alternative than this, but not required.

Reply:
https://github.com/napari/superqt/blob/main/src/superqt/utils/_throttler.py#L138-L140 
no, because cancel does not stop the timer. It only switches the binary flag.

Reply:
Just curious, why does it matter if the timer keeps running if it does nothing when it times out? (Which is what cancel() does)

Reply:
Qt may keep objects alive if events need to proceed. So the started timer may keep the parent object alive until the timer is finished. But this is only my guess, I have no valid proof.